hail/src/main/scala/is/hail/asm4s/CodeBuilder.scala

```diff
@@ -99,6 +99,26 @@ trait CodeBuilderLike {
 
   def whileLoop(c: Code[Boolean], emitBody: => Unit): Unit = whileLoop(c, _ => emitBody)
 
+  def forLoop(setup: => Unit, cond: Code[Boolean], incr: => Unit, emitBody: (CodeLabel) => Unit): Unit = {
```

Comment:
Looks good! I kinda wish there were a way to make it obvious from the method signature what the CodeLabel argument was here---it looks like it's essentially so things like conditions have an "after" label to jump to?

Reply:
I wish I could make it more obvious too. The label argument is `continue`. There's currently no `break`, though it can be easily simulated with an extra label.

```scala
val Lbreak = CodeLabel()
cb.whileLoop(cond, {
     // do stuff
     cb.ifx(cond2, cb.goto(Lbreak))
     // do more stuff
})
cb.define(Lbreak)
```